test/icareFhirMessaging.test.js

```diff
@@ -0,0 +1,73 @@
+const testBundle = require('./fixtures/message-bundle.json');
+const { checkMessagingClient, postExtractedData, getMessagingClient } = require('../src/icareFhirMessaging');
+
+const mockMessagingClient = {
+  authorize: jest.fn(),
+  canSendMessage: jest.fn(),
+  processMessage: jest.fn(),
+};
+
+// Utility for flattening error values
+function flattenErrorValues(errorValues) {
+  return _.flatten(Object.values(errorValues));
+}
+
+describe('icareFhirMessaging', () => {
+  describe('getMessagingClient', () => {
+    it('should throw error when awsConfig not provided in config', () => {
+      expect(() => getMessagingClient({})).toThrowError('config file is missing `awsConfig` field, which is required to create a messagingClient instance');
+    });
+    it('should return a messaging Client when the config is valid', () => {
+      const mClient = getMessagingClient(testConfig);
+      expect(() => getMessagingClient(testConfig)).not.toThrowError();
+      expect(mClient).toBeInstanceOf(MessagingClient);
+    });
+  });
+
+  describe('checkMessagingClient', () => {
+    it('should throw error when messaging client cannot send message', async () => {
+      mockMessagingClient.canSendMessage.mockReturnValue(false);
+      await expect(checkMessagingClient(mockMessagingClient)).rejects.toThrowError('The server does not provide the "system/$process-message" scope.');
+    });
+
+    it('should not throw error when messaging client can send message', async () => {
+      mockMessagingClient.canSendMessage.mockReturnValue(true);
+      await expect(checkMessagingClient(mockMessagingClient)).resolves.not.toThrowError();
+    });
+
+    it('should throw error when messaging client cannot authorize.', async () => {
+      const errorMsg = 'cannot authorize';
+      mockMessagingClient.authorize.mockImplementation(() => {
+        throw new Error(errorMsg);
+      });
+      await expect(checkMessagingClient(mockMessagingClient)).rejects.toThrowError(`Could not authorize messaging client - ${errorMsg}`);
+      mockMessagingClient.authorize.mockReset();
+    });
+  });
+
+  describe('postExtractedData', () => {
+    it('should post data successfully when supplied appropriate message bundles', async () => {
+      mockMessagingClient.canSendMessage.mockReturnValue(true);
+      // postExtractedData expects an araray of bundles, one for each patient
+      const arrayOfBundles = [testBundle.testBundle, testBundle];
+      const { successfulMessagePost, messagingErrors } = await postExtractedData(mockMessagingClient, arrayOfBundles);
+      expect(successfulMessagePost).toEqual(true);
+      const flattenedErrorValues = flattenErrorValues(messagingErrors);
+      expect(flattenedErrorValues).toEqual([]);
+      expect(mockMessagingClient.processMessage).toHaveBeenCalledTimes(arrayOfBundles.length);
+    });
+
+    it('should fail to post data when messaging client cannot process message', async () => {
```

Comment:
While I was looking into the unhandled promise rejections, I was running this test by itself, and it looks like we might need to mock the `canSendMessage` response in the test as well so that it can run independently of the one above it. So in this test or maybe a `beforeEach` in this describe block, we should include this:

```
mockMessagingClient.canSendMessage.mockReturnValue(true);
```

Reply:
Should be addressed